tsdb/mapper.go

```diff
@@ -114,185 +112,192 @@ func (lm *SelectMapper) Open() error {
 	}
 	lm.tx = tx
 
-	if s, ok := lm.stmt.(*influxql.SelectStatement); ok {
-		stmt, err := lm.rewriteSelectStatement(s)
-		if err != nil {
-			return err
-		}
-		lm.selectStmt = stmt
-		lm.rawMode = (s.IsRawQuery && !s.HasDistinct()) || s.IsSimpleDerivative()
-	} else {
-		return lm.openMeta()
-	}
-
-	// Set all time-related parameters on the mapper.
-	lm.queryTMin, lm.queryTMax = influxql.TimeRangeAsEpochNano(lm.selectStmt.Condition)
-
-	if !lm.rawMode {
-		if err := lm.initializeMapFunctions(); err != nil {
-			return err
-		}
-
-		// For GROUP BY time queries, limit the number of data points returned by the limit and offset
-		d, err := lm.selectStmt.GroupByInterval()
-		if err != nil {
-			return err
-		}
-		lm.intervalSize = d.Nanoseconds()
-		if lm.queryTMin == 0 || lm.intervalSize == 0 {
-			lm.numIntervals = 1
-			lm.intervalSize = lm.queryTMax - lm.queryTMin
+	if err := func() error {
```

Comment:
This line and the conditional block at the bottom are the only changes in this PR.